models/challenges.js

```diff
@@ -35,6 +36,36 @@ const fetchChallenges = async () => {
   }
 }
 
+/**
+ * Fetch the <user object> from participants array
+ * @param {Array} participants
+ * @returns {Promise<challengesModel|Array>}
+ */
+const fetchParticipantsData = async (participants) => {
+  try {
+    const fetchedparticipants = getParticipantsData(participants)
+    return fetchedparticipants
+  } catch (err) {
+    logger.error('Failed to get participated users', err)
+    throw err
+  }
+}
+
+/**
+ * Fetch the partipantsData from participants array
+ * @param {Array} participants
+ * @returns {Promise<participantsData|Array>}
+ */
+const getParticipantsData = async (participants) => {
+  return await Promise.all(participants.map(async (userId) => {
+    const { user } = await fetchUser({ userId })
+    return {
+      ...user,
+      phone: undefined,
+      email: undefined
+    }
+  }))
```

Comment:
I was saying it could be like `Promise.all(array of promises returns from another function)`